src/lib/fragments/AgGrid.react.js

```diff
@@ -469,9 +471,16 @@ export default class DashAgGrid extends Component {
 
     getRowData() {
         const newRowData = [];
-        this.state.gridApi.forEachNode((node) => {
-            newRowData.push(node.data);
-        });
+        const {openGroups} = this.state;
+        if (!isEmpty(openGroups)) {
+            this.state.gridApi.forEachLeafNode((node) => {
+                newRowData.push(node.data);
+            });
+        } else {
+            this.state.gridApi.forEachNode((node) => {
+                newRowData.push(node.data);
+            });
```

Comment:
Why is `!isEmpty(openGroups)` the condition to use `forEachLeafNode`? Would it work to just always use `forEachLeafNode` in this function?

Reply:
I changed it, all the tests passed, so I guess it is capable. 🤞